pkgs/native_assets_cli/lib/src/api/build_output.dart

```diff
@@ -77,18 +89,28 @@ abstract final class BuildOutput {
     Iterable<Uri>? dependencies,
     Map<String, Object>? metadata,
   }) =>
-      BuildOutputImpl(
+      HookOutputImpl(
         timestamp: timestamp,
         assets: assets?.cast<AssetImpl>().toList(),
         dependencies: Dependencies([...?dependencies]),
         metadata: Metadata({...?metadata}),
       );
 
   /// Adds [Asset]s produced by this build or dry run.
-  void addAsset(Asset asset);
+  ///
+  /// If the [linkInPackage] argument is specified, the asset will not be
+  /// bundled during the build step, but sent as input to the linking script
+  /// of the specified package, where it can be further processed and possibly
+  /// bundled.
+  void addAsset(Asset asset, {String? linkInPackage});
 
   /// Adds [Asset]s produced by this build or dry run.
-  void addAssets(Iterable<Asset> assets);
+  ///
+  /// If the [linkInPackage] argument is specified, the assets will not be
+  /// bundled during the build step, but sent as input to the linking script
```

Comment:
script -> hook

(Maybe ctrl+f for 'script' to find all other places where old terminology is floating.)